components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java

```diff
@@ -361,9 +361,9 @@ public int compare(IdentityProcessor identityProcessor1,
                                    IdentityProcessor identityProcessor2) {
 
                     if (identityProcessor1.getPriority() > identityProcessor2.getPriority()) {
-                        return 1;
-                    } else if (identityProcessor1.getPriority() < identityProcessor2.getPriority()) {
                         return -1;
+                    } else if (identityProcessor1.getPriority() < identityProcessor2.getPriority()) {
+                        return 1;
```

Comment:
According to API definition of Comparator this is what it says: "a negative integer, zero, or a positive integer as the first argument is less than, equal to, or greater than the second". Which means what is there in the code right now is correct isn't it?